src/shared-components/icon/index.js

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import iconFilePath from '../../svgs/icons.svg';
+import { COLORS } from '../../constants';
+
+const propTypes = {
+  iconName: PropTypes.string.isRequired,
+  width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  rotate: PropTypes.number,
+  fill: PropTypes.string,
+  inline: PropTypes.bool,
+  className: PropTypes.string,
+  duskId: PropTypes.string,
+};
+
+const defaultProps = {
+  className: '',
+  fill: COLORS.primary,
+  height: 15,
+  inline: false,
+  rotate: 0,
+  width: 15,
+};
+
+const Icon = ({
+  className,
+  fill,
+  height,
+  iconName,
+  inline,
+  rotate,
+  width,
+  duskId,
+  ...rest
+}) => {
+  // eslint-disable-next-line no-undef
+  const icon = window.__SVG_INLINE__
+    ? `#${iconName}`
+    : `${iconFilePath}#${iconName}`;
+  return (
+    <svg
+      width={width}
+      height={height}
+      fill={fill}
+      data-dusk-id={duskId}
+      css={`
+        display: ${inline ? 'inline-block' : 'block'};
+        transform: rotate(${rotate}deg);
+        color: ${fill};
```

Comment:
fill doesnt really mean the same thing as color, i think we should update one of the two

Reply:
so it does in certain cases. There was a movement once (started by me of course 😅) to set the fill on the definition of svgs to `currentColor` so that svgs can basically inherit the color set on the parent container rather than using `fill` css property which would only work on SVGs.

Looking at it now, we probably just want granular control over the fill by being more declarative on the icon set i guess? I'm just trying to think of times where the text + icon inside of a container would need to change color at the same time and having `currentColor` would be easier... Actually that might have been why we did it this way to begin with. May have been something on the signup flow where we have icons as radio buttons. It would also explain the transition on the color attribute.

Reply:
yea i like using currentColor. i just mean the exposed API should map to what is happening, so the prop should  probably be named `color` rather than fill

Reply:
For the sake of timeliness of converting things, I'd rather not change the interface and continue using `fill`. Also, `fill` isn't a stretch since all of our icons are SVG icons